lib/SPIRV/SPIRVRegularizeLLVM.cpp

```diff
@@ -413,6 +413,21 @@ bool SPIRVRegularizeLLVMBase::regularize() {
           if (isa<PossiblyExactOperator>(BO) && BO->isExact())
             BO->setIsExact(false);
         }
+
+        // FIXME: This is not valid handling for freeze instruction
+        if (auto *FI = dyn_cast<FreezeInst>(&II)) {
+          auto *V = FI->getOperand(0);
+          if (isa<UndefValue>(V)) {
+            if (V->getType()->isIntegerTy())
+              V = ConstantInt::get(V->getType(), 0);
+            else if (V->getType()->isFPOrFPVectorTy())
+              V = ConstantFP::get(V->getType(), 0.0);
```

Comment:
Would it be better to use `Constant::getNullValue` here, which would cover any type?

Reply:
Updated to use Constant::getNullValue.  Added testing for ptr type which needs this change.